polaris-core/src/main/java/org/apache/polaris/core/secrets/UserSecretsManager.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.polaris.core.secrets;
+
+import jakarta.annotation.Nonnull;
+import org.apache.polaris.core.entity.PolarisEntity;
+
+/**
+ * Manages secrets specified by users of the Polaris API, either directly or as an intermediary
+ * layer between Polaris and external secret-management systems. Such secrets are distinct from
+ * "service-level" secrets that pertain to the Polaris service itself which would be more statically
+ * configured system-wide. In contrast, user-owned secrets are handled dynamically as part of
+ * runtime API requests.
+ */
+public interface UserSecretsManager {
+  /**
+   * Persist the {@code secret} under a new URN {@code secretUrn} and return a {@code
+   * UserSecretReference} that can subsequently be used by this same UserSecretsManager to retrieve
+   * the original secret. The {@code forEntity} is provided for an implementation to optionally
+   * extract other identifying metadata such as entity type, name, etc., to store alongside the
+   * remotely stored secret to facilitate operational management of the secrets outside of the core
```

Comment:
If it's not expected that the USM will always persist the `PolarisEntity`, can we get a version of this method that doesn't take one?

Reply:
Even though it's optional for a USM implementation to use the `PolarisEntity`, it is non-optional for the *caller* to pass one in.

In general if at least some subset of receiver implementations require it, then all callers must provide it because receivers are injected impls.

If in the future we end up finding a scenario where a callsite *can't* provide the PolarisEntity, then I guess we need to think about adding a version of this method that doesn't take oine.

Reply:
From the other thread on this:
> URN should be self-sufficient to retrieve whatever is needed to be retrieved

If the URN really is sufficient to retrieve the secret, then let there be a method which takes a URN and returns a secret. If we can't do that, doesn't that imply that the URN by itself is never sufficient to retrieve a secret?

Reply:
From the rest of that sentence, the URN is sufficient to retrieve:

> whatever is needed to be retrieved from the external secret store.

The content that is retrieved from the external secret store may or may not be the original secret. We could have a method that takes a URN and returns "content from the external secret store" but that may or may not be interpretable by a caller.

Reply:
Per discussion, changing `PolarisEntity` to `PolarisEntityCore` which only contains the basic identifying attributes (id, name, type, parentId, catalogId, entityVersion) which should be sufficient for now.